llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCExpr.cpp

```diff
@@ -7,6 +7,9 @@
 //===----------------------------------------------------------------------===//
 
 #include "AMDGPUMCExpr.h"
+#include "GCNSubtarget.h"
```

Comment:
This include is also questionable layering-wise. MCTargetDesc is supposed to be a small(ish) library independent of the main Target library, but now it includes GCNSubtarget.h from the main target library, which pulls in AMDGPURegisterBankInfo.h which includes generated headers.

Reply:
My constraint was that I had to compute things like occupancy and extrasgpr usage in places where I couldn't assume MCExpr as resolvable and therefore had to delay computation through some of these custom MCExprs. Said computations were normally done through a lot of the GCNSubtarget calls unfortunately. I'll look into whether I can move some of these around to `Utils/AMDGPUBaseInfo`.